tket2/src/circuit/cost.rs

```diff
@@ -41,14 +42,29 @@ pub trait CostDelta:
 /// This is used to order circuits based on major cost first, then minor cost.
 /// A typical example would be CX count as major cost and total gate count as
 /// minor cost.
-#[derive(Clone, Copy, Default, PartialEq, Eq, PartialOrd, Ord, From)]
-pub struct MajorMinorCost<T = usize> {
-    major: T,
-    minor: T,
+pub type MajorMinorCost<T = usize> = LexicographicCost<T, 2>;
+
+impl<T> From<(T, T)> for MajorMinorCost<T> {
+    fn from((major, minor): (T, T)) -> Self {
+        Self([major, minor])
+    }
+}
+
+/// A cost that is ordered lexicographically.
+///
+/// An array of cost functions, where the first one is infinitely more important
+/// than the second, which is infinitely more important than the third, etc.
+#[derive(Clone, Copy, PartialEq, Eq, PartialOrd, Ord, From)]
```

Comment:
If you just want to allow any `Into<[T; N]>`,
```suggestion
impl<const N: usize, V, T> From<V> for LexicographicCost<T, N>
where
    V: Into<[T; N]>,
{
    fn from(v: V) -> Self {
        Self(v.into())
    }
}

/// A cost that is ordered lexicographically.
///
/// An array of cost functions, where the first one is infinitely more important
/// than the second, which is infinitely more important than the third, etc.
#[derive(Clone, Copy, PartialEq, Eq, PartialOrd, Ord)]
```